### ISSUE TITLE
VoiceOver doesn't announce day when focused on date in the DatePickerController

### ISSUE THREAD

**Issue by bansalr-microsoft** [microsoft/fluentui-apple#498]
### Environment Information

- **Platform**:
  - [ ] iOS
  - [X] macOS
- **Package version(s)**: Fluent UI 0.2.3
- **Xcode and OS versions**: Xcode 12.2, Mac OS 11.2.3

### Please provide a reproduction of the bug:

**Pre requisites:**
Turn on Voice over

**Repro steps:**
1. Open the DatePickerController 
2. Navigate to a day in the DatePickerController popup and observe voice over announcement.

#### Actual behavior:

When focused on any date in 'Pick a date and time' popup, observe voiceover announces as date format(2nd of November 2020).VoiceOver doesn't announce day.

#### Expected behavior:

When focused on any date in 'Pick a date and time' popup, observe VoiceOver should announce as day along with date format(Monday, 2nd of November 2020).

#### User Impact:
Visually impaired users who rely on screenreader will have to navigate to column again to know the day of focused date.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/applications affected: Microsoft To Do

**Comment by chripear** [comment]
ADO Ref: [To-Do App #41636](https://alexanderplatz.visualstudio.com/To-Do/_workitems/edit/41636)
MAS Ref: [MAS 4.2.1 - Object Information](https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc=%7ba12f3779-bdf1-40c2-bbcc-0e314ea94858%7d)
